src/gennylib/src/context.cpp

```diff
@@ -91,8 +91,7 @@ ActorVector WorkloadContext::_constructActors(const Cast& cast,
     } catch (const std::out_of_range&) {
         std::ostringstream stream;
         stream << "Unable to construct actors: No producer for '" << name << "'." << std::endl;
-        cast.streamProducersTo(stream);
```

Comment:
Is this method necessary here? It was adding garbage characters to the stream, and it seems like the error message here is sufficient enough. I couldn't understand why it was printing out garbage, and adding null checks to see that the values existed didn't eliminate the issue.

Reply:
I wouldn't expect it to lead to printing out garbage. I'm happy to take a look over your shoulder.

This may be unrelated but - I think `Cast::streamProducersTo()` should be using `_producers` [rather than `globalCast().getProducers()`](https://github.com/mongodb/genny/blob/b0992e276d8fb8f1c1f72bc62001a1d434617171/src/gennylib/src/Cast.cpp#L34) given that those two necessarily aren't the same for testing purposes.

Reply:
Strangely it only printed out garbage on my machine but not on Max's. In any case I changed it to _producers and that fixed the issue.